src/admin/resolvers/mutations/ApprovedItem.ts

```diff
@@ -178,6 +179,59 @@ export async function updateApprovedItem(
   return approvedItem;
 }
 
+/**
+ * A targeted update operation that only updates an approved item's authors data.
+ * Used to backfill authors for legacy curated items.
+ *
+ * @param parent
+ * @param data
+ * @param context
+ */
+export async function updateApprovedItemAuthors(
+  parent,
+  { data },
+  context: IContext
+): Promise<ApprovedItem> {
+  // Check if the user can perform this mutation
+  if (!context.authenticatedUser.canWriteToCorpus()) {
+    throw new AuthenticationError(ACCESS_DENIED_ERROR);
+  }
+  // To be able to delete authors associated with a corpus item, we first need
+  // to get the internal (integer) id for the story. This means doing a DB query
+  // to fetch the entire object.
+  const existingItem = await getApprovedItemByExternalId(
+    context.db,
+    data.externalId
+  );
+
+  // Remove the old author(s) from the DB records before we run the update function
+  // Note that we don't expect any authors to be present for any items this mutation
+  // will be run for, but it's a good idea to be thorough anyway.
+  if (existingItem) {
```

Comment:
if there is no `existingItem`, should we halt processing and throw an error? based on the data set we're working with, i can't see this ever happening, BUT if we checking for it anyway, we may as well add in an `else`, yeah?

Reply:
I'm not sure why it was set up this way (I think I wrote this??? :)), but I bet it was a TypeScript limitation, whereby Prisma could return a `null` for an approved item potentially and TS was not having it. Removing the brackets and using optional chaining instead works, too. I don't think there's a need for an `if` statement here.